codegenerator/cli/src/rescript_types.rs

```diff
@@ -19,6 +19,40 @@ impl RescriptTypeDeclMulti {
         Self(type_declarations)
     }
 
+    pub fn to_rescript_schema(&self) -> String {
+        let mut sorted: Vec<RescriptTypeDecl> = vec![];
+        let mut registered: HashSet<String> = HashSet::new();
+
+        let type_decls_with_deps: Vec<(RescriptTypeDecl, Vec<String>)> = self
+            .0
+            .iter()
+            .map(|decl| {
+                let deps = decl.type_expr.dependencies();
+                (decl.clone(), deps)
+            })
+            .collect();
+
+        // Not the most optimised algorithm, but it's simple and works:
+        // Iterate over the type declarations and add them to
+        // the sorted list if all their dependencies already added - repeat
+        while sorted.len() < type_decls_with_deps.len() {
+            for (decl, deps) in &type_decls_with_deps {
+                if !registered.contains(&decl.name)
+                    && deps.iter().all(|dep| registered.contains(dep))
+                {
+                    sorted.push(decl.clone());
+                    registered.insert(decl.name.clone());
+                }
+            }
+        }
```

Comment:
Nice, I can think of a few use cases for sorting deps like this. The alternative is just to use a `rec` definition with `and` between each one then you don't have to care about ordering.

Reply:
> Nice, I can think of a few use cases for sorting deps like this. The alternative is just to use a rec definition with and between each one then you don't have to care about ordering.

It only works when the value is a function. I tried it first, but didn't work.

> Actually with that in mind, can there ever be a cyclical/recursive dependency?

Thankfully, they are not supported in Fuel. Otherwise the code didn't work

Reply:
Sweet 👍🏼